src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/tools/PStack.java

```diff
@@ -81,6 +81,9 @@ public void run(PrintStream out, Debugger dbg) {
             out.println("can't print deadlock information: " + exp);
          }
 
+         VMLocksPrinter vmLocksPrinter = new VMLocksPrinter(out);
+         vmLocksPrinter.printVMLocks();
```

Comment:
It would be best to catch all exceptions here and continue on after a warning message. We do the same in the loop below as we iterate over each thread. This is a safety net in case some structures are currently in an inconsistent state.